.github/workflows/cicd.yaml

```
@@ -100,16 +104,25 @@ jobs:
           #- manuelt-vedlikehold
     steps:
       - uses: actions/checkout@v3
+        with:
+          fetch-depth: 5
+      - name: fix deploy sha for merge commit
+        # if the current sha is a merge commit we want the sha that was on the PR head before merge (the sha of the second parent)
+        run: |
+          echo "DEPLOY_SHA=$(git show ${{ github.sha }}^2 --pretty=format:'%H' --no-patch || git rev-parse ${{ github.sha }})" >> $GITHUB_ENV
```

```
          echo "DEPLOY_SHA=$(git show ${{ github.sha }}^2 --pretty=format:'%H' --no-patch || git rev-parse ${{ github.sha }})" >> $GITHUB_ENV
```
```suggestion
          echo "DEPLOY_SHA=$(git show ${{ github.sha }}^2 --pretty=format:'%H' --no-patch || echo ${{ github.sha }})" >> $GITHUB_ENV
```
`github.sha` vil jeg tro alltid er sha-en, så man kan bare returnere den som den er?

---

Ytterligere:
```suggestion
          echo "DEPLOY_SHA=$(git rev-parse ${{ github.sha }}^2 || echo ${{ github.sha }})" >> $GITHUB_ENV
```

---

> Ytterligere:

Dette fungerer ikke da rev-parse outputer sha^2 dersom det feiler. derfor jeg bruker show

---

> ```
>           echo "DEPLOY_SHA=$(git show ${{ github.sha }}^2 --pretty=format:'%H' --no-patch || git rev-parse ${{ github.sha }})" >> $GITHUB_ENV
> ```
> 
> `github.sha` vil jeg tro alltid er sha-en, så man kan bare returnere den som den er?

ja dette er i praksis det samme, men jeg synes echo i echo smakte litt rart. derfor gitt jeg for rev-parse 🤷

---

Skjønner. Koden blir jo kort og fin med `git show`! Jeg var bekymret for at `fetch-depth: 5` naivt hentet de 5 siste commitene, og at vi derfor kunne gå glipp av `${{ github.sha }}^2`, avhengig av hvordan git-historikken så ut. Men ser at `fetch-depth` henter fra alle parents. Så dette virker som en lettlest og robust løsning!

(Jeg synes `echo` eller `printf` hadde vært bedre enn `git rev-parse`, altså. Hvis jeg leser koden om et år, så ville jeg igjen dobbeltsjekket i terminalen om den ikke gjør noe annet enn å printe det ut igjen, sittet og lurt på om det er fordi `github.sha` kan være noe commit-ish i noen tilfeller som må resolves, for så å sitte å lure på om det er noe fundamentalt jeg har gått glipp av.)

---

godt poeng, mindre forvirrende med echo eller printf

---

07c01ef659e00f912f154713b716608613f49d85